### ISSUE TITLE
Kinesis Stream costs are only estimated for us-east-1

### ISSUE THREAD

**Issue by stanch** [infracost/infracost#2731]
Using the latest version with Kinesis Stream support (0.10.30), I’ve noticed that all my unit prices related to Kinesis Stream are $0.

[In the code](https://github.com/infracost/infracost/blob/master/internal/resources/aws/kinesis_stream.go#L244), the attribute filters are set to values like `Storage-ShardHour`.

However, I’ve confirmed in the price database (self-hosted) that the name is actually different for every region:
```sql
select region, attributes->'usagetype' as usagetype
from products
where "productFamily" = 'Kinesis Streams' and attributes->>'usagetype' like '%Storage-ShardHour';
```

```
     region     |        usagetype
----------------+--------------------------
 af-south-1     | "AFS1-Storage-ShardHour"
 ap-east-1      | "APE1-Storage-ShardHour"
 ap-northeast-1 | "APN1-Storage-ShardHour"
 ap-northeast-2 | "APN2-Storage-ShardHour"
 ap-northeast-3 | "APN3-Storage-ShardHour"
 ...
 us-east-1      | "Storage-ShardHour"
 us-east-2      | "USE2-Storage-ShardHour"
 us-gov-east-1  | "UGE1-Storage-ShardHour"
 us-gov-west-1  | "UGW1-Storage-ShardHour"
 us-west-1      | "USW1-Storage-ShardHour"
 us-west-2      | "USW2-Storage-ShardHour"
```

Further, I’ve modified the database to test out the theory that the filter is incorrect:
```sql
update products
set attributes = jsonb_set(attributes, '{usagetype}', '"Storage-ShardHour"', false)
where attributes->>'usagetype' = 'EUC1-Storage-ShardHour';
```

After this change, I got the correct unit price.

I believe the attribute filters should use regular expressions, like [here](https://github.com/infracost/infracost/blob/master/internal/resources/aws/instance.go#L252), instead of static strings.

**Comment by tim775** [comment]
@stanch Thanks for finding this and thanks for the great report!  Let me know if you are interested in send a PR for the fix yourself, otherwise I will open it up to the reset of the team/community.  Either way we should get this fixed in the next release.

**Comment by tim775** [comment]
@stanch Thanks again for the report.  The fix will go out in the 10.31 release.  If you're using the github/lab app, it should be fixed now.

**Comment by stanch** [comment]
Many thanks @tim775!

**Comment by aliscott** [comment]
@stanch thanks for reporting this 🙏 The fix is now released in [v0.10.31](https://github.com/infracost/infracost/releases/tag/v0.10.31).
